app/src/main/java/edu/grinnell/appdev/grinnelldirectory/User.java

```diff
@@ -1,37 +1,66 @@
 package edu.grinnell.appdev.grinnelldirectory;
-// package conventions?
 
 import android.content.SharedPreferences;
 import android.content.SharedPreferences.Editor;
 
 import android.content.Context;
 
+import com.google.gson.annotations.SerializedName;
+
 /**
- * Created by alsrfo on 2/10/17.
+ * The User class represents the user of the app.
+ * It handles storage of username and password in shared preferences.
+ * User objects are immutable. They will not change when the user's credentials are changed using
+ * <code>saveCredentials</code> or <code>deleteCredentials</code>.
  */
 
 public class User {
 
-    private final static String SHARED_PREFERENCES = "user";
+    private final static String PREF_USER = "PREF_USER";
     private final static String USERNAME = "username";
     private final static String PASSWORD = "password";
 
-    public static String getUsername(Context context) {
-        return getSharedPreferences(context).getString(USERNAME, null);
+    @SerializedName("un")
+    private final String mUsername;
+
+    @SerializedName("pw")
+    private final String mPassword;
+
+    public User(String username, String password) {
+        mUsername = username;
+        mPassword = password;
     }
 
-    public static String getPassword(Context context) {
-        return getSharedPreferences(context).getString(PASSWORD, null);
+    /**
+    * Get a User object with username and password from shared preferences
+    * @param context context of the activity that calls this method
+    * @return User object with saved username and password
+    */
+    public static User getUser(Context context) {
+        SharedPreferences preferences = getSharedPreferences(context);
+        String username = preferences.getString(USERNAME, null);
+        String password = preferences.getString(PASSWORD, null);
+        return new User(username, password);
     }
 
+    /**
+     * Determine whether a user is currently logged in
+     * @param context context of the activity that calls this method
+     * @return whether a user is logged in
+     */
     public static boolean isLoggedIn(Context context) {
         SharedPreferences preferences = getSharedPreferences(context);
         String username = preferences.getString(USERNAME, null);
         String password = preferences.getString(PASSWORD, null);
         return username != null && password != null;
     }
 
-    /* Sets new username and password if they are valid */
+    /**
+     * Save the username and password in shared preferences
+     * @param context context of the activity that calls this method
+     * @param username the user's new username
+     * @param password the user's new password
+     */
     public static void saveCredentials(Context context, String username, String password) {
```

Comment:
Just to be on the safe side here, I'd just check to make sure `username` and `password` are not null or empty. Exit the function immediately if either or both are.